enhancers.go

```diff
@@ -2,29 +2,325 @@ package main
+	}
+
+	// Add the cronjob to the fingerprint
+	sentryEvent.Fingerprint = append(sentryEvent.Fingerprint, "job", jobObj.Name)
+
+	// Add the cronjob to the tag
+	setTagIfNotEmpty(scope, "job_name", object.GetName())
+	jobObj.ManagedFields = []metav1.ManagedFieldsEntry{}
+	metadataJson, err := prettyJson(jobObj.ObjectMeta)
+	if err == nil {
+		scope.SetContext("job", sentry.Context{
+			"Metadata": metadataJson,
+		})
+	}
+
+	// Add breadcrumb with cronjob timestamps
+	scope.AddBreadcrumb(&sentry.Breadcrumb{
+		Message:   fmt.Sprintf("Created job %s", object.GetName()),
+		Level:     sentry.LevelInfo,
+		Timestamp: object.GetCreationTimestamp().Time,
+	}, breadcrumbLimit)
+
+	return nil
+}
+
+func cronjobPodEnhancer(ctx context.Context, scope *sentry.Scope, object metav1.Object, sentryEvent *sentry.Event) error {
+
+	cronjobObj, ok := object.(*batchv1.CronJob)
+	if !ok {
+		return errors.New("failed to cast object to CronJob object")
+	}
+
+	// Set the context for corresponding slug monitor
+	scope.SetContext("Monitor", sentry.Context{
+		"Slug": object.GetName(),
+	})
+
+	// Add the cronjob to the fingerprint
+	sentryEvent.Fingerprint = append(sentryEvent.Fingerprint, "cronjob", cronjobObj.Name)
+
+	// Add the cronjob to the tag
+	setTagIfNotEmpty(scope, "cronjob_name", object.GetName())
+	cronjobObj.ManagedFields = []metav1.ManagedFieldsEntry{}
+	metadataJson, err := prettyJson(cronjobObj.ObjectMeta)
+	if err == nil {
+		scope.SetContext("cronjob", sentry.Context{
+			"Metadata": metadataJson,
+		})
+	}
+
+	// Add breadcrumb with cronjob timestamps
+	scope.AddBreadcrumb(&sentry.Breadcrumb{
+		Message:   fmt.Sprintf("Created cronjob %s", object.GetName()),
+		Level:     sentry.LevelInfo,
+		Timestamp: object.GetCreationTimestamp().Time,
+	}, breadcrumbLimit)
+
+	return nil
+}
+
+func replicaSetPodEnhancer(ctx context.Context, scope *sentry.Scope, object metav1.Object, sentryEvent *sentry.Event) error {
+
+	replicasetObj, ok := object.(*appsv1.ReplicaSet)
+	if !ok {
+		return errors.New("failed to cast object to ReplicaSet object")
+	}
+
+	// Add the cronjob to the fingerprint
+	sentryEvent.Fingerprint = append(sentryEvent.Fingerprint, "replicaset", replicasetObj.Name)
+
+	// Add the cronjob to the tag
+	setTagIfNotEmpty(scope, "replicaset_name", object.GetName())
+	replicasetObj.ManagedFields = []metav1.ManagedFieldsEntry{}
+	metadataJson, err := prettyJson(replicasetObj.ObjectMeta)
+	if err == nil {
+		scope.SetContext("replicaset", sentry.Context{
+			"Metadata": metadataJson,
+		})
+	}
+
+	// Add breadcrumb with cronjob timestamps
+	scope.AddBreadcrumb(&sentry.Breadcrumb{
+		Message:   fmt.Sprintf("Created replicaset %s", object.GetName()),
+		Level:     sentry.LevelInfo,
+		Timestamp: object.GetCreationTimestamp().Time,
+	}, breadcrumbLimit)
+
+	return nil
+}
+
+func deploymentPodEnhancer(ctx context.Context, scope *sentry.Scope, object metav1.Object, sentryEvent *sentry.Event) error {
```

Comment:
Look like *PodEnhancer functions can be refactored (do they have to be Pod-related?), but it's ok to do it in a separate PR.